### PR TITLE
[DO NOT MERGE] Documentation update review

### PR DESCRIPTION
**Pull request by deusebio**
@grobbie  @marcoppenheimer @zmraul @a-velasco

### CODE REVIEW

File: docs/how-to/h-cluster-migration.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ In short, MirrorMaker runs as a distributed service on the new cluster, and cons
 - An existing Kafka cluster to migrate from
 - A bootstrapped Juju VM machine cloud running Charmed Kafka to migrate to
     - A tutorial on how to set-up a Charmed Kafka deployment can be found as part of the [Charmed Kafka Tutorial](/t/charmed-kafka-tutorial-overview/10571)
-- `yq` - https://github.com/mikefarah/yq
+    - `yq` - https://github.com/mikefarah/yq
     - `snap install yq --channel=v3/stable`
```
Comment (a-velasco) [inline docs/how-to/h-cluster-migration.md:26]:
```suggestion
    - CLI file processor tool `yq` - https://github.com/mikefarah/yq
    - `snap install yq --channel=v3/stable`
```

Doesn't this make more sense as its own bullet point instead of a sub-point of "A boostrapped Juju VM machine..."?

File: docs/how-to/h-deploy.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+```shell
+$ juju deploy zookeeper --channel 3/edge -n <zookeeper-units>
+$ juju deploy kafka --channel 3/edge -n <kafka-units>
+```
+
+After this, it is necessary to connect them:
+```shell
+$ juju relate kafka zookeeper
+```
```
Comment (a-velasco) [inline docs/how-to/h-deploy.md:68]:
```suggestion
After this, it is necessary to [integrate](https://juju.is/docs/juju/relation) them:
```shell
$ juju integrate kafka zookeeper
```
```

File: docs/how-to/h-deploy.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+## Deploy Charmed Kafka and Charmed ZooKeeper
+
+The Kafka and ZooKeeper charms can both be deployed as follows:
+
+```shell
+$ juju deploy zookeeper --channel 3/edge -n <zookeeper-units>
+$ juju deploy kafka --channel 3/edge -n <kafka-units>
+```
+
+After this, it is necessary to connect them:
```
Comment (a-velasco) [inline docs/how-to/h-deploy.md:65]:
```suggestion
After this, it is necessary to [integrate](https://juju.is/docs/juju/relation) them:
```

File: docs/how-to/h-deploy.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+```shell
+$ juju relate kafka zookeeper
```
Comment (a-velasco) [inline docs/how-to/h-deploy.md:67]:
```suggestion
$ juju integrate kafka zookeeper
```

File: docs/how-to/h-deploy.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+to be used to manage the Kafka cluster (either internally or externally). 
+
+To create an admin user, deploy the [Data Integrator Charm](https://charmhub.io/data-integrator) with 
+`extra-user-roles` set to `admin`
+
+```shell
+juju deploy data-integrator --channel stable --config topic-name=test-topic --config extra-user-roles=admin
+```
+
+and relate to the Kafka charm
```
Comment (a-velasco) [inline docs/how-to/h-deploy.md:94]:
```suggestion
and integrate with the Kafka charm
```

File: docs/how-to/h-deploy.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+```shell
+juju relate data-integrator kafka
```
Comment (a-velasco) [inline docs/how-to/h-deploy.md:97]:
```suggestion
juju integrate data-integrator kafka
```

File: docs/how-to/h-manage-units.md [hunk 7]
```diff
@@ -1,55 +1,102 @@
-# How to deploy and manage units
+# How to manage units
 
-## Basic usage
+Apache Kafka ships with `bin/*.sh` commands to do various administrative tasks, among which
```
Comment (a-velasco) [inline docs/how-to/h-manage-units.md:3]:
```suggestion
Apache Kafka ships with `bin/*.sh` commands to do various administrative tasks such as:
```

File: docs/how-to/h-manage-units.md [hunk 8]
```diff
@@ -1,55 +1,102 @@
...
-### Scaling down
-To scale down the charm, use `juju remove-unit` command.
 ```shell
-juju remove-unit <unit_name>
+juju add-unit kafka -n <num_brokers_to_add>
 ```
 
-Even when scaling multiple units at the same time, the charm uses a rolling restart sequence to make sure the cluster stays available and healthy during the operation.
\ No newline at end of file
+For more information on how to manage units, please refer to [this](https://juju.is/docs/juju/manage-units)
```
Comment (a-velasco) [inline docs/how-to/h-manage-units.md:89]:
```suggestion
For more information on how to manage units, please refer to the [Juju documentation](https://juju.is/docs/juju/manage-units)
```

File: docs/how-to/h-manage-units.md [hunk 9]
```diff
@@ -1,55 +1,102 @@
...
+The `client.properties` file can be generated by replacing relevant information in the 
+one available on the brokers, e.g. 
+
 ```
+BOOTSTRAP_SERVERS=$(juju run data-integrator/leader get-credentials --format yaml | yq .kafka.endpoints )
+USERNAME=$(juju run data-integrator/leader get-credentials --format yaml | yq .kafka.username )
+PASSWORD=$(juju run data-integrator/leader get-credentials --format yaml | yq .kafka.password )
+```
+
+## Replication and Scaling
```
Comment (a-velasco) [inline docs/how-to/h-manage-units.md:80]: I think this should be moved up so it's the first section. It's the default instructions on how to scale with juju, in case that's all the user is looking for, and then more complex management details can follow for more advanced users. 
What do you think?

File: docs/index.md [hunk 10]
```diff
@@ -1,17 +1,18 @@
 ## Charmed Kafka Documentation
 
-The Charmed Kafka Operator delivers automated operations management from day 0 to day 2 on the [Apache Kafka](https://kafka.apache.org) event streaming platform. It is an open source, end-to-end, production ready data platform on top of cloud native technologies.
+Charmed Kafka is an open-source operator that makes it easier to operate Apache Kafka with built-in support for enterprise features. 
 
-This operator charm comes with features such as:
-- Fault-tolerance, replication, scalability and high-availability out-of-the-box.
-- SASL/SCRAM auth for Broker-Broker and Client-Broker authentication enabled by default.
-- Access control management supported with user-provided ACL lists.
+Charmed Kafka is build on top of [Juju](https://juju.is/) and it simplifies the deployment, scaling, design and management of Apache Kafka in production in a reliable way. In addition, you can use the operator to manage your Kafka clusters with automation capabilities. It also offers replication, TLS, password rotation, easy-to-use application integration, and monitoring.
```
Comment (a-velasco) [inline docs/index.md:5]:
```suggestion
Charmed Kafka is built on top of [Juju](https://juju.is/) and reliably simplifies the deployment, scaling, design, and management of Apache Kafka in production. Additionally, you can use the charm to manage your Kafka clusters with automation capabilities. It also offers replication, TLS, password rotation, easy-to-use application integration, and monitoring.
```

File: docs/index.md [hunk 11]
```diff
@@ -1,17 +1,18 @@
...
+Charmed Kafka operates Apache Kafka on physical, Virtual Machines (VM) and other wide range of cloud and cloud-like environments, including AWS, Azure, OpenStack and VMWare. 
 
-The Kafka Operator uses the latest upstream Kafka binaries released by the The Apache Software Foundation that comes with Kafka, made available using the [`charmed-kafka` snap ](https://snapcraft.io/charmed-kafka) distributed by Canonical.
+Charmed Kafka is a solution designed and developed for helping organizations and 
+DB administrators automating Apache Kafka operations management from day 0 to day 2, across multiple cloud environments and substrates.
 
-As currently Kafka requires a paired ZooKeeper deployment in production, this operator makes use of the [ZooKeeper Operator](https://github.com/canonical/zookeeper-operator) for various essential functions.
+Aside from a VM operator, Charmed Kafka K8s provides support for Kubernetes environments. For more information, please refer to the Charmed Kafka K8s documentation [here](/t/charmed-kafka-k8s-documentation/10296).
```
Comment (a-velasco) [inline docs/index.md:11]:
```suggestion
[note]
Canonical has also developed the [Charmed Kafka K8s operator]((/t/charmed-kafka-k8s-documentation/10296)) to support Kafka in Kubernetes environments.
[/note]
```

File: docs/index.md [hunk 12]
```diff
@@ -1,17 +1,18 @@
 ## Charmed Kafka Documentation
```
Comment (a-velasco) [inline docs/index.md:1]: To keep a consistent home page structure across charms, order would be:
1. Title
2. Description
3. "In this documentation"
4. "Project and community"
5. License

File: docs/reference/r-file-system-path.md [hunk 13]
```diff
@@ -0,0 +1,36 @@
+# File System Path
+
+In the following table, we summarize some of the most relevant file path used in
```
Comment (a-velasco) [inline docs/reference/r-file-system-path.md:3]:
```suggestion
In the following table, we summarize some of the most relevant file paths used in 
```

File: docs/reference/r-file-system-path.md [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+| /var/snap/charmed-kafka/common/var/log/kafka/  | Application Logging files generated by the Kafka daemon process. These files are written by the workload, but they may be read by other components to provide monitoring (e.g. grafana and/or charm). | (owned and managed by `snap_daemon`)          |
+| /var/snap/charmed-kafka/common/var/lib/kafka/  | Raw data stored persistently by Kafka during its operations. The files are written and managed by Kafka only.                                                                                         | (owned and managed by `snap_daemon`)          |
+
+
+External storage is used for storing persistent raw data, and it is  
+mounted at `/var/snap/charmed-kafka/common/var/lib/kafka/<storage-id>`, 
+where `<storage-id>` is a progressive number allowing multiple storage units
+for providing both horizontal scalability and provide IO parallelization to 
+enhance throughput.
```
Comment (a-velasco) [inline docs/reference/r-file-system-path.md:21]: This is a loong sentence for a brain to process, and I found it hard to understand what `<storage-id>` actually is.
Is it a unique ID that gets generated automatically when an external storage unit is mounted? (and the advantages of this are that it allows for multiple storage units, which provides scalability and parallelization etc etc?)

I would provide a simple definition of `<storage-id>` and separate the last part into its own sentence, like "Multiple storage units provides horizontal scalability and provides IO parallelization to enhance throughput."

File: docs/how-to/h-deploy.md (hunk 3 above)
Comment (deusebio) [reply docs/how-to/h-deploy.md:65]: yes, I'm afraid we may have a bit of "relate" all over the place right now (even in other docs that are not included in this PR). I'll try to do a search and replace, but in one commit to hopefully address this consistently

File: docs/reference/r-snap-entrypoints.md [hunk 15]
```diff
@@ -0,0 +1,40 @@
+# Charmed Kafka Snap Entrypoints
+
+The Snap entrypoints wrap the Kafka Distribution Bash scripts and they make sure
+that these scripts are run with the correct environment settings 
+(configuration files, logging files, etc). Here below, we provide a reference
+table for the mapping between entrypoints and wrapped bash script.
```
Comment (a-velasco) [inline docs/reference/r-snap-entrypoints.md:6]:
```suggestion
Snap entrypoints wrap the Kafka Distribution Bash scripts and make sure
that they run with the correct environment settings (configuration files, logging files, etc). 

Below is a reference table for the mapping between entrypoints and wrapped bash script:
```

**Review by a-velasco — APPROVED** [review]
LGTM! I really like the new reference tables.

File: docs/index.md [hunk 16]
```diff
@@ -1,17 +1,18 @@
...
+Charmed Kafka is an open-source operator that makes it easier to operate Apache Kafka with built-in support for enterprise features. 
 
-This operator charm comes with features such as:
-- Fault-tolerance, replication, scalability and high-availability out-of-the-box.
-- SASL/SCRAM auth for Broker-Broker and Client-Broker authentication enabled by default.
-- Access control management supported with user-provided ACL lists.
+Charmed Kafka is build on top of [Juju](https://juju.is/) and it simplifies the deployment, scaling, design and management of Apache Kafka in production in a reliable way. In addition, you can use the operator to manage your Kafka clusters with automation capabilities. It also offers replication, TLS, password rotation, easy-to-use application integration, and monitoring.
+Charmed Kafka operates Apache Kafka on physical, Virtual Machines (VM) and other wide range of cloud and cloud-like environments, including AWS, Azure, OpenStack and VMWare.
```
Comment (grobbie) [inline docs/index.md:6]:
```suggestion
Charmed Kafka operates Apache Kafka on physical systems, Virtual Machines (VM) and a wide range of cloud and cloud-like environments, including AWS, Azure, OpenStack and VMWare. 
```

File: docs/index.md [hunk 17]
```diff
@@ -1,17 +1,18 @@
...
-- SASL/SCRAM auth for Broker-Broker and Client-Broker authentication enabled by default.
-- Access control management supported with user-provided ACL lists.
+Charmed Kafka is build on top of [Juju](https://juju.is/) and it simplifies the deployment, scaling, design and management of Apache Kafka in production in a reliable way. In addition, you can use the operator to manage your Kafka clusters with automation capabilities. It also offers replication, TLS, password rotation, easy-to-use application integration, and monitoring.
+Charmed Kafka operates Apache Kafka on physical, Virtual Machines (VM) and other wide range of cloud and cloud-like environments, including AWS, Azure, OpenStack and VMWare. 
 
-The Kafka Operator uses the latest upstream Kafka binaries released by the The Apache Software Foundation that comes with Kafka, made available using the [`charmed-kafka` snap ](https://snapcraft.io/charmed-kafka) distributed by Canonical.
+Charmed Kafka is a solution designed and developed for helping organizations and 
+DB administrators automating Apache Kafka operations management from day 0 to day 2, across multiple cloud environments and substrates.
```
Comment (grobbie) [inline docs/index.md:9]:
```suggestion
administrators to automate Apache Kafka operations from day 0 to day 2, across multiple cloud environments and platforms.
```

File: docs/index.md [hunk 18]
```diff
@@ -1,17 +1,18 @@
...
 
-This operator charm comes with features such as:
-- Fault-tolerance, replication, scalability and high-availability out-of-the-box.
-- SASL/SCRAM auth for Broker-Broker and Client-Broker authentication enabled by default.
-- Access control management supported with user-provided ACL lists.
+Charmed Kafka is build on top of [Juju](https://juju.is/) and it simplifies the deployment, scaling, design and management of Apache Kafka in production in a reliable way. In addition, you can use the operator to manage your Kafka clusters with automation capabilities. It also offers replication, TLS, password rotation, easy-to-use application integration, and monitoring.
+Charmed Kafka operates Apache Kafka on physical, Virtual Machines (VM) and other wide range of cloud and cloud-like environments, including AWS, Azure, OpenStack and VMWare. 
 
-The Kafka Operator uses the latest upstream Kafka binaries released by the The Apache Software Foundation that comes with Kafka, made available using the [`charmed-kafka` snap ](https://snapcraft.io/charmed-kafka) distributed by Canonical.
+Charmed Kafka is a solution designed and developed for helping organizations and
```
Comment (grobbie) [inline docs/index.md:8]:
```suggestion
Charmed Kafka is a solution designed and developed for helping ops teams and 
```

File: docs/index.md [hunk 19]
```diff
@@ -1,17 +1,18 @@
...
 
-As currently Kafka requires a paired ZooKeeper deployment in production, this operator makes use of the [ZooKeeper Operator](https://github.com/canonical/zookeeper-operator) for various essential functions.
+Aside from a VM operator, Charmed Kafka K8s provides support for Kubernetes environments. For more information, please refer to the Charmed Kafka K8s documentation [here](/t/charmed-kafka-k8s-documentation/10296).
 
-The Charmed Kafka operator comes in two flavours to deploy and operate Kafka on [physical/virtual machines](https://github.com/canonical/kafka-operator) and [Kubernetes](https://github.com/canonical/kafka-k8s-operator). Both offer features such as replication, TLS, password rotation, and easy to use integration with applications. The Charmed Kafka Operator meets the need of deploying Kafka in a structured and consistent manner while allowing the user flexibility in configuration. It simplifies deployment, scaling, configuration and management of Kafka in production at scale in a reliable way.
+Charmed Kafka is developed and supported by [Canonical](https://ubuntu.com/), as part of its commitment to
```
Comment (grobbie) [inline docs/index.md:15]:
```suggestion
Charmed Kafka is developed and supported by [Canonical](https://canonical.com/), as part of its commitment to 
```

**Review by grobbie — CHANGES_REQUESTED** [review]
left some comments

File: docs/how-to/h-deploy.md [hunk 20]
```diff
@@ -0,0 +1,111 @@
...
+
+Once all the units show as `active|idle` in the `juju status` output, the deployment 
+should be ready to be used. 
+
+## (Optional) Create an external admin users
+
+After being deployed, the Kafka cluster is *secure by default* and, as a consequence, 
+it won't expose any external listener. 
+In fact, ports are only opened when client applications are related, also 
+depending on the protocols to be used. Please refer to [this table](TODO) for
```
[the remark this answers is not in the document]
Comment (deusebio) [reply docs/how-to/h-deploy.md:81]: yes, this will be updated once we start publishing this content into Discourse

File: docs/how-to/h-deploy.md [hunk 21]
```diff
@@ -0,0 +1,111 @@
...
+and relate to the Kafka charm
+
+```shell
+juju relate data-integrator kafka
+```
+
+To retrieve authentication information such as the username, password, etc. use
+
+```shell
+juju run data-integrator/leader get-credentials
```
[the remark this answers is not in the document]
Comment (deusebio) [reply docs/how-to/h-deploy.md:103]: it should! if there is one unit only, that's obviously the leader

File: docs/how-to/h-enable-monitoring.md [hunk 22]
```diff
@@ -7,8 +7,8 @@ Additionally, the charm provides integration with the [Canonical Observability S
 
 Deploy cos-lite bundle in a Kubernetes environment. This can be done by following the
 [deployment tutorial](https://charmhub.io/topics/canonical-observability-stack/tutorials/install-microk8s).
-Since the Charmed Kafka Operator is deployed on a machine environment, it is needed to offer the endpoints
-of the COS relations. The [offers-overlay](https://github.com/canonical/cos-lite-bundle/blob/main/overlays/offers-overlay.yaml)
+Since the Charmed Kafka Operator is deployed on a machine environment, it is 
+needed to offer the endpoints of the COS relations. The [offers-overlay](https://github.com/canonical/cos-lite-bundle/blob/main/overlays/offers-overlay.yaml)
```
[the remark this answers is not in the document]
Comment (deusebio) [reply docs/how-to/h-enable-monitoring.md:11]: I believe I have tested this on k8s actually. I will double check on VM

File: docs/how-to/h-manage-units.md [hunk 23]
```diff
@@ -1,55 +1,102 @@
...
+for a full list of the bash commands available in Kafka distributions. Also, you can 
+use `--help` argument for printing a short summary of the argument for a given 
+bash command. 
 
-After this, it is necessary to connect them:
-```shell
-$ juju relate kafka zookeeper
-```
+The most important commands are also exposed via the [Charmed Kafka snap](https://snapcraft.io/charmed-kafka), 
+accessible via `charmed-kafka.<command>`. Please refer to [this table](TODO) for
```
[the remark this answers is not in the document]
Comment (deusebio) [reply docs/how-to/h-manage-units.md:41]: as before.

**Review by zmraul — COMMENTED** [review]
@deusebio @a-velasco fyi related to changing command `relate -> integrate`. 
```
[~]$ j remove-integration kafka zookeeper                                 
ERROR juju: "remove-integration" is not a juju command. See "juju --help".
                                                                          
Did you mean:                                                             
        remove-relation                                                   
```
Should we keep it as `relate` until `remove-integration` exists?

**Comment by deusebio** [comment]
I'm not sure there is any plan in Juju to also have `remove-relation`. I would agree that it would make things more consistent, but if there is no actual plan, we should not constraint one to another. Either way, I suppose we should move away from `juju relate` as this appears with deprecated warnings in the logs. 

Anyhow, I don't mind too much. Up to @a-velasco to make a call such that we keep documentation consistent across DP.

**Comment by deusebio** [comment]
Actually I just found [this](https://matrix.to/#/!eNCNzcteYUDDYpStsu:ubuntu.com/$4Yz37m_KG8J7yf0N8lxXG8zkeCzHPfiioVaz99cgmKs?via=ubuntu.com&via=matrix.org) just spot on. 

TLDR `remove-relation` still makes sense. 

`integrate` is the verb, `relation` is the noun. So the action/verb is `integrating`, but whenever we refer to something, it is a "relation", e.g. "Integrating is the action of creating a relation`

**Comment by deusebio** [comment]
superseded by #175
